src/components/campaignguide/ScenarioStepComponent/ResolutionStepComponent.tsx

```diff
@@ -42,9 +33,9 @@ export default function ResolutionStepComponent({ step }: Props) {
       { (!!resolution.text || resolution.steps.length > 0) && (
         <View style={space.marginTopM}>
           <View style={{ ...space.marginSideM, display: 'flex', flexDirection: 'row', alignItems: 'center' }}>
-            { hasDS && resolution.narration && (
-              <Icon name="play-circle-outline" type="material" onPress={playNarrationPressed} />
-            ) }
+            <CheckDissonantVoicesComponent narration={resolution.narration}>
+              <NarrationStatusButton narration={resolution.narration!}/>
```

Comment:
I'd rather avoid using the `!` if possible.  This could be setup as a `continuation` children, i.e. define children as `(narration: Narration) => React.ReactNode | React.ReactNode[]`. I have a few components written like this, though I think the preferred construct would be to define this as a hook rather than an HOC.


Something like
```narration = useNarration(resolution.narration)``` 
The hook could check the state/constant, and return `narration` or `undefined` appropriately, and then the fucntion below could be written as a simple `!!narration && <NarrationStatusButton narration={narration} />`

Reply:
That sounds like a better way to do it. I'll look into it.